operator/src/main/java/oracle/kubernetes/operator/Main.java

```diff
@@ -929,31 +1025,57 @@ public NextAction onSuccess(Packet packet, CallResponse<V1NamespaceList> callRes
       String intialResourceVersion = getInitialResourceVersion(result);
       List<String> nsList = getExistingNamespaces(result);
       
-      Set<String> namespacesToStart = new TreeSet<>(targetNamespaces);
-      for (String ns : targetNamespaces) {
-        if (!nsList.contains(ns)) {
-          try (LoggingContext stack = LoggingContext.setThreadContext().namespace(ns)) {
-            LOGGER.warning(MessageKeys.NAMESPACE_IS_MISSING, ns);
+      Set<String> namespacesToStart;
+      if (selectionStrategy.isRequireList()) {
+        namespacesToStart = new TreeSet<>(nsList);
+        String regexp = selectionStrategy.getRegExp();
+        if (regexp != null) {
+          namespacesToStart = namespacesToStart.stream().filter(
+                  Pattern.compile(regexp).asPredicate()).collect(Collectors.toSet());
```

Comment:
can the regexp be valid, and `Pattern.compile(regexp)` may throw `PatternSyntaxException`? Perhaps it has been validated elsewhere that I missed.